src/main/java/uk/nhs/hee/tis/trainee/forms/mapper/LtftMapper.java

```diff
@@ -21,23 +21,42 @@
 
 package uk.nhs.hee.tis.trainee.forms.mapper;
 
+import static org.mapstruct.InjectionStrategy.CONSTRUCTOR;
 import static org.mapstruct.MappingConstants.ComponentModel.SPRING;
+import static uk.nhs.hee.tis.trainee.forms.dto.enumeration.LifecycleState.UNSUBMITTED;
 
+import jakarta.annotation.Nullable;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
 import java.util.List;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.Setter;
 import org.mapstruct.InheritInverseConfiguration;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.springframework.beans.factory.annotation.Autowired;
 import uk.nhs.hee.tis.trainee.forms.dto.LtftAdminSummaryDto;
 import uk.nhs.hee.tis.trainee.forms.dto.LtftAdminSummaryDto.LtftAdminPersonalDetailsDto;
 import uk.nhs.hee.tis.trainee.forms.dto.LtftFormDto;
 import uk.nhs.hee.tis.trainee.forms.dto.LtftSummaryDto;
 import uk.nhs.hee.tis.trainee.forms.model.LtftForm;
+import uk.nhs.hee.tis.trainee.forms.model.content.CctChange;
+import uk.nhs.hee.tis.trainee.forms.model.content.LtftContent;
 
 /**
  * A mapper to convert between LTFT related entities and DTOs.
  */
-@Mapper(componentModel = SPRING, uses = TemporalMapper.class)
-public interface LtftMapper {
+@Mapper(componentModel = SPRING, uses = TemporalMapper.class, injectionStrategy = CONSTRUCTOR)
+public abstract class LtftMapper {
+
+  private static final int NOTICE_PERIOD_DAYS = 112; // 16 weeks.
+
+  @Getter
+  @Setter(onMethod_ = @Autowired)
```

Comment:
Google incoming... 😂

Reply:
I was hoping to be able to have an abstract `getTemporalMapper()` which would return the `TemporalMapper` from the generated `LtftMapperImpl`, but unfortunately mapstruct doesn't support that. So this is a bit of lombok magic to inject it here as well as in to the constructor on the implementation.

It does lead to some awkward initialisation in tests, but it is made a little easier with the `CONSTRUCTOR` injection strategy avoiding our previous reflection-based approach to initialising nested mappers.